rest_framework/schemas/openapi.py

```diff
@@ -39,16 +40,19 @@ def get_schema(self, request=None, public=False):
         Generate a OpenAPI schema.
         """
         self._initialise_endpoints()
+        components_schemas = {}
 
         # Iterate endpoints generating per method path operations.
-        # TODO: …and reference components.
         paths = {}
         _, view_endpoints = self._get_paths_and_endpoints(None if public else request)
         for path, method, view in view_endpoints:
             if not self.has_view_permissions(path, method, view):
                 continue
 
             operation = view.schema.get_operation(path, method)
+            components = view.schema.get_components(path, method)
+            components_schemas.update(components)
```

Comment:
I'm wondering if we should check that the new components are not overriding existing components here. That could prevent unexpecting behavior: if a user has multiple serializers with the same name.

However, someone may wants to override a component set previously. What's your opinion on this @carltongibson  ?

Reply:
I don't think it's for an individual inspector (AutoSchema) to worry about overriding previously set components. 

**Maybe** we could log a warning if the name was already taken. That's likely a mistake. 

Folks should be overriding get_component_name() if they have similarly named serializers, I guess.

Reply:
Thanks for the feedback! I added a warning if different components have the same name.